README.md

```diff
@@ -930,6 +930,85 @@ Output:
 
 </details>
 
+<details>
+<summary>Reflection</summary>
+
+Blueprint classes may be reflected on to inspect their views, fields, and associations. Extensions often make use of this ability.
+
+```ruby
+class WidgetBlueprint < Blueprinter::Base
+  fields :name, :description
+  association :category, blueprint: CategoryBlueprint
+
+  view :extended do
+    field :price
+    association :parts, blueprint: WidgetPartBlueprint
+  end
+end
+
+# A Hash of views keyed by name
+views = WidgetBlueprint.reflections
+views.keys
+=> [:default, :extended]
+
+# Hashes of fields and associations, keyed by name
+fields = views[:default].fields
+assoc = views[:default].associations
+
+# Get info about a field
+fields[:description].name
+fields[:description].display_name
+fields[:description].options
+
+# Get info about an association
+assoc[:category].name
+assoc[:category].display_name
+assoc[:category].blueprint
+assoc[:category].view
+assoc[:category].options
+```
+</details>
+
+<details>
+<summary>Extensions</summary>
+
+Blueprinter offers an extension system to hook into and modify certain behavior.
+
+```ruby
+Blueprinter.configure do |config|
+  config.extensions << MyExtension.new
+  config.extensions << OtherExtension.new
+end
+```
+
+The following class implements all extension hooks. If you have an idea for a new one, open an issue or PR.
+
+```ruby
+class ExampleExtension < Blueprinter::Extension
+  #
+  # Called eary during "render", this method receives the object to be rendered and
```

Comment:
I think we missed this in the top level comment of `Blueprinter::Extension` as well. 

```suggestion
  # Called early during "render", this method receives the object to be rendered and
```

Reply:
As an aside, it looks like we're essentially just copying the base class here. Would it be helpful to either:

1. Link out to the code itself, and provide a list of "hooks" that are currently available
2. Enumerate the list of "hooks" and add an example using them.